src/CodeGen_Internal.cpp

```diff
@@ -560,55 +555,50 @@ Expr lower_round_to_nearest_ties_to_even(const Expr &x) {
 }
 
 namespace {
-bool get_md_int(llvm::Metadata *value, int64_t &result) {
+std::optional<int64_t> get_md_int(llvm::Metadata *value) {
     if (!value) {
-        return false;
+        return {};
```

Comment:
style q: if this preferable to returning `std::nullopt`? I would have thought the latter to be better, but whatever...

Reply:
It felt natural to me because it looks like the empty set, and "nullopt" didn't seem like it communicated anything useful (there's no null pointer involved, so why "null"). But I don't feel strongly and can change it if you think it would be an improvement.

Reply:
Returning `nullopt` is a good habbit, as it's ambiguous for struct types that take an empty constructor. What is `std::optional<std::string> test = {}`? I honestly don't know... https://godbolt.org/z/G683WKTxa Godbolt shows it will take the nullopt, instead of the no-args-constructed object.